### ISSUE TITLE
FutureWarnings all over the place

### ISSUE THREAD

**Issue by sammosummo** [Theano/Theano#6755]
I see warnings such as the following in numerous places:

```
/Users/samuelrobertmathias/miniconda3/envs/sammosummo.github.io/lib/python3.8/site-packages/theano/tensor/subtensor.py:2197: FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.
  rval = inputs[0].__getitem__(inputs[1:])
```

I realize Theano is no longer developed. Googling suggests this warning is something packages using numpy under the hood need fix themselves, so posting in case anyone sees.

**Comment by nouiz** [comment]
Make sure to use the github version of Theano. It contains fixes for newer NumPy from memory.

**Comment by 1fish2** [comment]
No doubt you've all seen these warnings like from pytest running the PyPI release `Theano==1.0.4`:

```
.../site-packages/theano/compat/__init__.py:16: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    from collections import (OrderedDict, MutableMapping as DictMixin,

.../site-packages/theano/misc/frozendict.py:9: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    class frozendict(collections.Mapping):
```

Is there a plan to release an update to PyPI for compatibility with Python 3.9, even if it needs a new PyPI name?

**Comment by twiecki** [comment]
@1fish2 Want to do a PR with those import fixes? I can release a new version on pypi for that.

**Comment by 1fish2** [comment]
@twiecki Yes! I'll give it a go.

**Comment by twiecki** [comment]
@1fish2 Any progress?

**Comment by 1fish2** [comment]
Yes. Progress in the sense that I can start on it today after crossing off top priority tasks. [Getting to Python 3 is harder than it seems, partly due to problems in installing libraries.] From the docs this looks straightforward...

**Comment by twiecki** [comment]
Closed by #6759.
